apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -508,7 +508,7 @@ export class WorkshopForm extends React.Component {
       // Temporarily hiding CSF 201 subject from the workshop dashboard
       // until the pilot is over in winter 2018
       const included_subjects = Subjects[this.state.course]
-        .filter(subject => subject !== "Deep Dive Workshop");
+        .filter(subject => subject !== "Deep Dive");
```

Comment:
could this instead use the `SUBJECT_CSF_201` constant?

Reply:
I'm not sure how to access the ruby constant in JavaScript... Could we ship this fix to unblock DTP (since we don't want the subject going out visible to everyone). And I can look into it now how to use a constant here instead?

Reply:
These are defined in [shared constants](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/shared_constants/pd/shared_workshop_constants.rb), so the values are actually available in JS-land in `@cdo/apps/generated/pd/sharedWorkshopConstants`, but not the constant names :(

The generated JS format is like this, which doesn't really help:
```jsx
export const Subjects = {
  "Exploring Computer Science": [
    "Phase 2 in-person",
    "Unit 3 - HTML",
    "Unit 4 - Scratch",
    "Unit 5 - Data",
    "Unit 6 - Robotics",
    "Phase 4: Summer wrap-up"
  ],
  "CS in Algebra": [
    "Phase 2 in-person",
    "Phase 3: Academic Year Development"
  ],
  "CS in Science": [
    "Phase 2: Blended Summer Study",
    "Phase 3 - Semester 1",
    "Phase 3 - Semester 2"
  ],
  "CS Principles": [
    "5-day Summer",
    "1-day Academic Year, Units 1 and 2",
    "1-day Academic Year, Unit 3",
    "1-day Academic Year, Unit 4 + Explore Prep",
    "1-day Academic Year, Unit 5 + Create Prep",
    "2-day Academic Year, Units 1 to 3",
    "2-day Academic Year, Units 4 and 5 + AP Prep",
    "Code.org TeacherCon",
    "Code.org Facilitator Weekend"
  ],
  "CS Discoveries": [
    "5-day Summer",
    "1-day Academic Year, Units 1 and 2",
    "1-day Academic Year, Unit 3",
    "1-day Academic Year, Units 4 and 5",
    "1-day Academic Year, Unit 6",
    "2-day Academic Year, Units 1 to 3",
    "2-day Academic Year, Units 4 to 6",
    "Code.org TeacherCon",
    "Code.org Facilitator Weekend"
  ],
  "CS Fundamentals": [
    "Intro Workshop",
    "Code.org Facilitator Weekend"
  ]
};
```

Reply:
We could set up an Ruby Hash instead of an array of constants, then export that in the [generateSharedConstants](https://github.com/code-dot-org/code-dot-org/blob/4a2f390dcd7300fe69f4c6b85c9839fbc0b9c5a2/apps/script/generateSharedConstants.rb#L85-L92) script to make common keys available in JS. It would require some rewriting on both sides though.

Reply:
Oh, if the override is temporary, a constant is almost certainly unnecessary. I just wanted to defend against the situation where we change the name here again sometime far in the future and have to remember to update this filter again.

Reply:
I completely agree!